parsl/executors/workqueue/executor.py

```diff
@@ -759,6 +768,28 @@ def submit(self, func, resource_specification, *args, **kwargs):
 
         return fu
 
+    def _construct_worker_command(self):
+        worker_command = 'work_queue_worker'
+        if self.project_password_file:
+            worker_command += ' --password {}'.format(self.project_password_file)
+        if self.project_name:
+            worker_command += ' -M {}'.format(self.project_name)
+        else:
+            worker_command += ' {} {}'.format(self.hostname, self.port)
```

Comment:
I am not quite sure about this `else`. I think `hostname` and `port` are required for `work_queue_worker`, no? With `project_name` being set, are these two optional?

Reply:
That's correct, if project_name is set, hostname and port are not used. With project_name, the master reports to a catalog of project_names -> host:port  (by default to the one at Notre Dame), from which the workers can get the host:port information.